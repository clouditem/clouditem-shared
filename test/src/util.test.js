const {unit_test} = require('../util.js'),
  under_test = require('../../src/util.js');

describe('promise_value', () => {

  const circular_request = {},
    tests = [
      {name: 'api_key in query successfull', successfull: true, value: 'api_key', in: ['query'], request: {query: {api_key: {}}}},
      {name: 'api_key in body and query prefers first', successfull: true, value: 'api_key', in: ['body', 'query'], request: {query: {api_key: 'from query'}, body: {api_key: 'from body'}}},
      {name: 'api_key in query and body prefers first reversed', successfull: true, value: 'api_key', in: ['query', 'body'], request: {query: {api_key: 'from query'}, body: {api_key: 'from body'}}},
      {name: 'error with circular request structure does show normal error', successfull: false, value: 'api_key', in: ['body'], request: circular_request.body = circular_request},
      {name: 'no request', successfull: false, value: 'api_key', in: ['query'], request: undefined},
      {name: 'empty request', successfull: false, value: 'api_key', in: ['query'], request: {}},
      {name: 'no matching in', successfull: false, value: 'api_key', in: ['query'], request: {body: {api_key: {}}, params: {}}},
      {name: 'no matching in multiple', successfull: false, value: 'api_key', in: ['query', 'params'], request: {body: {api_key: {}}, params: {}}},
    ],
    make_test = test_description => unit_test()
      .given(() => test_description.request)
      .when(req => under_test.promise_value(test_description.value)
        .in(test_description.in)
        .of_request(req))
      .is_rejected(!test_description.successfull)
      .then(res => expect(res).toMatchSnapshot())
      .done();
  tests.forEach(test_description => test(
    `${test_description.name}`,
    make_test(test_description)
  ));

});
describe('promise_presence', () => {

  const tests = [
      {name: 'undefined', successfull: false, input: {name: 'test_input', value: undefined}, assertion: res => res.toMatchSnapshot()},
      {name: 'false', successfull: false, input: {name: 'test_input', value: false}, assertion: res => res.toMatchSnapshot()},
      {name: 'zero', successfull: false, input: {name: 'test_input', value: 0}, assertion: res => res.toMatchSnapshot()},
      {name: 'null', successfull: false, input: {name: 'test_input', value: null}, assertion: res => res.toMatchSnapshot()},
      {name: 'empty String', successfull: false, input: {name: 'test_input', value: ''}, assertion: res => res.toMatchSnapshot()},
      {name: 'name undefined', successfull: false, input: {name: undefined, value: undefined}, assertion: res => res.toMatchSnapshot()},
      {name: 'filled String', successfull: true, input: {name: 'test_input', value: 'bla'}, assertion: res => res.toEqual('bla')},
      {name: 'empty Object', successfull: true, input: {name: 'test_input', value: {}}, assertion: res => res.toEqual({})},
    ],
    make_test = test_description => unit_test()
      .given(() => test_description.input)
      .when(input => under_test.promise_presence(input.name, input.value))
      .is_rejected(!test_description.successfull)
      .then(res => test_description.assertion(expect(res)))
      .done();
  tests.forEach(test_description => test(
    `${test_description.name}`,
    make_test(test_description)
  ));

});
