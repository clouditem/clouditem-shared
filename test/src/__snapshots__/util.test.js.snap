// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promise_presence empty String 1`] = `""`;

exports[`promise_presence false 1`] = `false`;

exports[`promise_presence name undefined 1`] = `[Error: expected undefined to be present]`;

exports[`promise_presence null 1`] = `[Error: expected test_input to be present]`;

exports[`promise_presence undefined 1`] = `[Error: expected test_input to be present]`;

exports[`promise_presence zero 1`] = `0`;

exports[`promise_value api_key in [query] successfull 1`] = `
Object {
  "promised_params": Object {
    "api_key": Object {},
  },
  "query": Object {
    "api_key": Object {},
  },
}
`;

exports[`promise_value api_key in body and query prefers first 1`] = `
Object {
  "body": Object {
    "api_key": "from body",
  },
  "promised_params": Object {
    "api_key": "from body",
  },
  "query": Object {
    "api_key": "from query",
  },
}
`;

exports[`promise_value api_key in query and body prefers first reversed 1`] = `
Object {
  "body": Object {
    "api_key": "from body",
  },
  "promised_params": Object {
    "api_key": "from query",
  },
  "query": Object {
    "api_key": "from query",
  },
}
`;

exports[`promise_value api_key in query successfull 1`] = `
Object {
  "promised_params": Object {
    "api_key": Object {},
  },
  "query": Object {
    "api_key": Object {},
  },
}
`;

exports[`promise_value empty request 1`] = `[Error: Request does not contain value [api_key] in any of the following fields: [query]]`;

exports[`promise_value error with circular request structure does show normal error 1`] = `[Error: Request does not contain value [api_key] in any of the following fields: [body]]`;

exports[`promise_value no matching in 1`] = `[Error: Request does not contain value [api_key] in any of the following fields: [query]]`;

exports[`promise_value no matching in multiple 1`] = `[Error: Request does not contain value [api_key] in any of the following fields: [query,params]]`;

exports[`promise_value no request 1`] = `[Error: expected Request to be present]`;
